test/e2e/go.mod

```diff
@@ -0,0 +1,12 @@
+module github.com/tetratelabs/getenvoy/e2e
```

Comment:
this is the new go.mod for e2e tests. As you can see archiver/v3 is here and we can also add anything we want/need without fear of accidentally bloating the binary. cc @musaprg @mathetake